src/tests/ftest/object/same_key_different_value.py

```diff
@@ -87,47 +74,40 @@ def test_single_to_array_value(self):
                 if i == 0:
                     try:
                         # write array value data to same keys, expected to fail
-                        self.container.write_an_array_value(array_value_data,
-                                                            dkey, akey, obj,
-                                                            obj_cls=1)
+                        self.container.write_an_array_value(
+                            array_value_data, dkey, akey, obj, obj_cls=1)
 
                         # above line is expected to return an error,
                         # if not fail the test
-                        self.fail("Array value write to existing single value"
-                                  + " key should have failed\n")
+                        self.fail(
+                            "Array value write to existing single value key should have failed")
 
                     # should fail with -1001 ERR
-                    except DaosApiError as excp:
-                        if "-1001" not in str(excp):
-                            self.log.info(excp)
-                            self.fail("Should have failed with -1001 error"
-                                      + " message, but it did not\n")
+                    except DaosApiError as error:
+                        if "-1001" not in str(error):
+                            self.log.error(error)
+                            self.fail("Expected write to fail with -1001")
 
                 # test case 2 and 3
-                elif i == 1 or 2:
+                else:
                     try:
                         # punch the keys
                         obj.punch_akeys(0, dkey, [akey])
                         obj.punch_dkeys(0, [dkey])
 
-                        if aggregation is True:
+                        if aggregation:
                             # trigger aggregation
                             self.container.aggregate(self.container.coh, 0)
 
-                        # write to the same set of keys under same object
-                        # with array value type
-                        self.container.write_an_array_value(array_value_data,
-                                                            dkey, akey, obj,
-                                                            obj_cls=1)
+                        # write to the same set of keys under same object with array value type
+                        self.container.write_an_array_value(
```

Comment:
(pylint-no-member) Instance of 'list' has no 'write_an_array_value' member
(pylint-no-member) Instance of 'TestContainer' has no 'write_an_array_value' member